example/web-avatar-client/test/mmlParsing.test.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import collectionData from "../src/collection.json";
+import { findAssetsInCollection } from "../src/findAssetsInCollection";
+import { parseMMLDescription } from "../src/parseMMLDescription";
+
+import {
+  threeNestedMCharacters,
+  threeRogueMModels,
+  twoInvalidlyWrappedMModel,
+  twoInvalidlyWrappedMModelOnInvalidMCharacterWith2ValidMModels,
+  twoRedundantMCharacterswithThreeMModelsEach,
+  validMCharacter,
+  validMCharacterWithOneInvalidMModel,
+  validMCharacterWithTwoInvalidMModels,
+} from "./test-data";
+import { extractNumberFromErrorMessage } from "./test-utils";
+
+describe("WebAvatarClient Test Utils", () => {
+  it("extractNumberFromErrorMessage should extract the number correctly from the error message", () => {
+    let err = "ignoring 1 <m-model> tag that were found..";
+    expect(extractNumberFromErrorMessage(err)).toBe(1);
+    err = "ignoring 4 <m-model> tags that were found..";
+    expect(extractNumberFromErrorMessage(err)).toBe(4);
+    err = "ignoring 8 <m-character> tags that were found...";
+    expect(extractNumberFromErrorMessage(err)).toBe(8);
+  });
+});
+
+describe("WebAvatarClient MML Parsing", () => {
```

Comment:
I'd suggest simplifying these tests to be an object contents equality check rather than pattern matching. It would be easier to see what the expected outcome was and would be less brittle, even if it were more code.